scripts/azureml-assets/azureml/assets/copy_assets.py

```diff
@@ -31,9 +35,31 @@ def copy_asset(asset_config: assets.AssetConfig,
         # Skip a released version
         return None
 
+    if check_previous_release:
+        # Skip if previous version was not released
+        tag = assets.validate_new_release(asset_config, release_directory_root)
+
+        if tag is not None:

```

Comment:
Should this code be left here or should it be a function inside of pin_image_versions? Secondly, should a retry also be done?

Reply:
I think it makes sense to reuse the manifest retrieval code in `pin_image_versions`, if you don't think it's a bad idea to take a dependency on it. Then you'd get the retry logic, etc. And then here you'd just decide what to do the response (404 or otherwise).